web/views/shared_view.ex

```diff
@@ -144,10 +144,10 @@ defmodule CoursePlanner.SharedView do
   end
 
   def form_datetime(form, field, opts \\ []) do
-    value =
+    value = Map.get(form.source.changes, field) ||
```

Comment:
Can you extract this in a function? This `case` after a `||` looks very confusing.